### ISSUE TITLE
voting on chain

### ISSUE THREAD

**Issue by monperrus** [KTH/programmable-society#15]

**Comment by monperrus** [comment]
[Analysis of blockchain solutions for e-voting: A systematic literature review](https://ieeexplore.ieee.org/abstract/document/9812616/)

**Comment by monperrus** [comment]
Pebble combines time lock encryption, zero-knowledge proofs, and blockchain technologies to enable self-tallying elections
https://www.pebble.vote/

**Comment by bbaudry** [comment]
Snapshot is a voting platform that allows DAOs, DeFi protocols, or NFT communities to vote easily and without gas fees.
https://docs.snapshot.org/readme
https://github.com/snapshot-labs

**Comment by monperrus** [comment]
B-Privacy: Defining and Enforcing Privacy in Weighted Voting
https://arxiv.org/pdf/2509.17871

TLDR: This work proposes a mechanism to boost B-privacy by noising voting tallies, which captures the economic cost to an adversary of bribing voters based on revealed voting tallies, and offers the first principled guidance on transparency-privacy tradeoffs in weighted-voting systems.

Abstract: In traditional, one-vote-per-person voting systems, privacy equates with ballot secrecy: voting tallies are published, but individual voters' choices are concealed. Voting systems that weight votes in proportion to token holdings, though, are now prevalent in cryptocurrency and web3 systems. We show that these weighted-voting systems overturn existing notions of voter privacy. Our experiments demonstrate that even with secret ballots, publishing raw tallies often reveals voters' choices. Weighted voting thus requires a new framework for privacy. We introduce a notion called B-privacy whose basis is bribery, a key problem in voting systems today. B-privacy captures the economic cost to an adversary of bribing voters based on revealed voting tallies. We propose a mechanism to boost B-privacy by noising voting tallies. We prove bounds on its tradeoff between B-privacy and transparency, meaning reported-tally accuracy. Analyzing 3,582 proposals across 30 Decentralized Autonomous Organizations (DAOs), we find that the prevalence of large voters ("whales") limits the effectiveness of any B-Privacy-enhancing technique. However, our mechanism proves to be effective in cases without extreme voting weight concentration: among proposals requiring coalitions of $\geq5$ voters to flip outcomes, our mechanism raises B-privacy by a geometric mean factor of $4.1\times$. Our work offers the first principled guidance on transparency-privacy tradeoffs in weighted-voting systems, complementing existing approaches that focus on ballot secrecy and revealing fundamental constraints that voting weight concentration imposes on privacy mechanisms.

**Comment by monperrus** [comment]
Kleros dispute resolution system
https://resources.cryptocompare.com/asset-management/682/1694772721553.pdf
